pkg/discovery/service.go

```diff
@@ -91,14 +95,47 @@ func (discovery *Discovery) Run(ctx context.Context) error {
 // We currently do not penalize nodes that are unresponsive,
 // but should in the future.
 func (discovery *Discovery) Refresh(ctx context.Context) error {
-	// TODO(coyle): make refresh work by looking on the network for new ndoes
 	nodes := discovery.kad.Seen()
 	for _, v := range nodes {
 		if err := discovery.cache.Put(ctx, v.Id, *v); err != nil {
 			return err
 		}
 	}
 
+	list, more, err := discovery.cache.Paginate(ctx, discovery.config.RefreshLimit, int64(offset))
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	// more means there are more rows to page through in the cache
+	if more == false {
+		offset = 0
+	} else {
+		offset = offset + len(list)
```

Comment:
this is racy, since offset is global. should offset be something like discovery.refreshOffset?

Reply:
Putting it on the struct 👍